cloudify_kubernetes/tasks.py

```diff
@@ -158,6 +158,60 @@ def _do_resource_status_check(resource_kind, response):
         else:
             ctx.logger.debug('status {0}'.format(status))
 
+    elif resource_kind == 'Deployment':
+        conditions = response['status']['conditions']
+        if conditions:
+            for condition in conditions:
+                if condition['type'] == 'Available':
+                    ctx.logger.debug('Deployment condition is Available')
+
+                elif condition['type'] == 'ReplicaFailure':
+                    raise NonRecoverableError(
+                        'Deployment condition is ReplicaFailure ,'
+                        'reason:{0}, message: {1}'
+                        ''.format(condition['reason'], condition['message']))
+
+                elif condition['type'] == 'Progressing':
+                    raise OperationRetry(
+                        'Deployment condition is Progressing')
+        else:
+            raise OperationRetry('Deployment condition is not ready yet')
+
+    elif resource_kind == 'PersistentVolumeClaim':
+        status = response['status']['phase']
+        if status in ['Pending', 'Available']:
+            raise OperationRetry(
+                'PersistentVolumeClaim status is {},'
+                'No persistent volume is detected'.format(status))
+
+        elif status == 'Bound':
+            ctx.logger.debug('PersistentVolumeClaim status is Bound')
+
+    elif resource_kind == 'PersistentVolume':
+        status = response['status']['phase']
+        if status in ['Bound', 'Available']:
+            ctx.logger.debug('PersistentVolume status is {0}'.format(status))
+
+        else:
+            raise OperationRetry(
+                'Unknown PersistentVolume status {0}'.format(status))
+
+    elif resource_kind in ['ReplicaSet', 'ReplicationController']:
+        ready_replicas = response['status'].get('ready_replicas')
+        replicas = response['status']['replicas']
+
+        if ready_replicas is None:
+            raise OperationRetry(
+                '{0} status not ready yet'.format(resource_kind))
+
+        elif ready_replicas != replicas:
+            raise OperationRetry(
+                'Only {0} of {1} replicas are ready'.format(
+                    ready_replicas, replicas))
+
+        elif ready_replicas == replicas:
+            ctx.logger.debug('All {0} replicas are ready now'.format(replicas))
+
```

Comment:
This last bunch of logic, I'm not quite sure about.

Reply:
For last statement I'm checking to see if all replica defined on the blueprint for k8s all are ready and running and provisioned from replicaset